### PR TITLE
notifier: Refactor and add client certificate authentification support.

### LINKED ISSUE #23: Add client cert auth to the HTTP notifier

### PR DESCRIPTION
**Pull request by Quentin-M**
Fixes #23

### CODE REVIEW

File: notifier/notifier.go [hunk 1]
```diff
@@ -24,150 +24,172 @@ import (
...
+	lockDuration        = time.Minute*8 + refreshLockDuration
 )
 
-// A HTTPNotifier dispatches notifications to an HTTP endpoint with POST requests
-type HTTPNotifier struct {
-	url string
+// A Notification represents the structure of the notifications that are sent by a Notifier.
+type Notification struct {
+	Name, Type string
+	Content    interface{}
```
Comment (jzelinskie) [inline notifier/notifier.go:47]: Eventually we should change this type to something like `json.Marshaller`.

File: notifier/notifier.go [hunk 2]
```diff
@@ -24,150 +24,172 @@ import (
...
+// A Notifier dispatches notifications to an HTTP endpoint.
+type Notifier struct {
+	lockIdentifier string
+	endpoint       string
+	client         *http.Client
+}
 
-	return notifier, nil
+// Config represents the configuration of a Notifier.
+// The certificates are optionnals and enables client certificate authentification.
```
Comment (jzelinskie) [inline notifier/notifier.go]: The certifications are optional and enable TLS client certificate authentication.

**Comment by Quentin-M** [comment]
Updated.

File: notifier/notifier.go [hunk 3]
```diff
@@ -24,150 +24,172 @@ import (
...
 }
 
-// Run pops notifications from the database, lock them, send them, mark them as
-// send and unlock them
-//
-// It uses an exponential backoff when POST requests fail
-func (notifier *HTTPNotifier) Run(st *utils.Stopper) {
-	defer st.End()
+// NewNotifier initializes a new Notifier from the specified configuration.
+func NewNotifier(cfg Config) *Notifier {
```
Comment (jzelinskie) [inline notifier/notifier.go]: You could just call this `New` to make it less redundant. It would look like `notifier.New()` then.

**Comment by jzelinskie** [comment]
rebase

**Comment by jzelinskie** [comment]
LGTM

**Pull request merged**